tests/samples: use integration_platforms more where it makes sense

nashif opened the pull request:
Use integration platforms to limit churn and build time in PR CI .

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests/samples: use integration_platforms more where it makes sense

Use integration platforms to limit churn and build time in PR CI .

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

gmarull commented on samples/subsys/display/lvgl/sample.yaml:
```diff
@@ -17,3 +17,5 @@ tests:
     modules:
       - lvgl
+    integration_platforms:
+      - mimxrt1064_evk
```
note: native_posix can be an option as well (with SDL display)

nashif replied:
yeah, tried this, but this requires 32bit SDL2 on the host and not everyone will have that.

gmarull replied:
so maybe `native_posix_64`? `libsdl2-dev` seems to be a Zephyr requirement: https://docs.zephyrproject.org/latest/develop/getting_started/index.html, but... it probably does not belong to getting started...

nashif committed:
tests: use integration_platforms where applicable

Use integration_platforms where coverage is provided using one or few
targets instead.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

cfriedt reviewed (approved):
LGTM
